holidays/countries/macau.py

```diff
@@ -0,0 +1,461 @@
+            dts_observed.add(self._add_chinese_birthday_of_buddha(tr("佛誕節")))
+
+            # Tung Ng Festival (Dragon Boat Festival).
+            dts_observed.add(self._add_dragon_boat_festival(tr("端午節")))
+
+            # The Day following Chong Chao (Mid-Autumn) Festival.
+            dts_observed.add(self._add_mid_autumn_festival_day_two(tr("中秋節翌日")))
+
+            # Chung Yeung Festival (Festival of Ancestors).
+            dts_observed.add(self._add_double_ninth_festival(tr("重陽節")))
+
+            # National Day of the People's Republic of China.
+            dts_observed.add(self._add_holiday_oct_1(tr("中華人民共和國國慶日")))
+
+            # The day following National Day of the People's Republic of China.
+            dts_observed.add(self._add_holiday_oct_2(tr("中華人民共和國國慶日翌日")))
+
+            # All Soul's Day.
+            dts_observed.add(self._add_all_souls_day(tr("追思節")))
+
+            # Immaculate Conception.
+            dts_observed.add(self._add_immaculate_conception_day(tr("聖母無原罪瞻禮")))
+
+            # Macao S.A.R. Establishment Day.
+            dts_observed.add(self._add_holiday_dec_20(tr("澳門特別行政區成立紀念日")))
+
+            # Winter Solstice.
+            dts_observed.add(self._add_holiday(tr("冬至"), self._winter_solstice_date))
+
+            # Christmas Eve.
+            dts_observed.add(self._add_christmas_eve(tr("聖誕節前日")))
+
+            # Christmas Day.
+            dts_observed.add(self._add_christmas_day(tr("聖誕節")))
+
+            if self._year >= 2018 and self._year not in {2022, 2023}:
+                # New Year's Eve.
+                self._add_new_years_eve(begin_time_label % self.tr("除夕"))
+
+        if self.observed:
+            self._populate_observed(dts_observed)
+
+    def _populate_subdiv_i_public_holidays(self):
+        # Decreto-Lei n.º 15/93/M - Moved Day of the Municipality of Ilhas from JUL 13 to NOV 30.
+        # Regulamento Administrativo n.º 4/1999 - Removed as a Public Holiday.
+        if self._year <= 1999:
+            # Day of the Municipality of Ilhas.
+            name = tr("海島市日")
+            if self._year <= 1992:
+                self._add_holiday_nov_30(name)
+            else:
+                self._add_holiday_jul_13(name)
+
+    def _populate_subdiv_m_public_holidays(self):
+        # Regulamento Administrativo n.º 4/1999 - Removed Macau City Day as a Public Holiday.
+        if self._year <= 1999:
+            # Macau City Day.
+            self._add_holiday_jun_24(tr("澳門市日"))
+
+    @property
+    def _winter_solstice_date(self) -> tuple[int, int]:
+        # This approximation is reliable for 1952-2099 years.
+        if (
+            (self._year % 4 == 0 and self._year >= 1988)
+            or (self._year % 4 == 1 and self._year >= 2021)
+            or (self._year % 4 == 2 and self._year >= 2058)
+            or (self._year % 4 == 3 and self._year >= 2091)
+        ):
+            day = 21
+        else:
+            day = 22
+        return DEC, day
```

Comment:
As Winter Solstice is already used in two countries, it's reasonable to move it to `ChineseCalendarHolidays` (as property and/or as `_add...` method, as for Ching Ming Festival).